proof/refine/ARM/SchedContextInv_R.thy

```diff
@@ -70,14 +70,65 @@ primrec sc_ctrl_inv_rel ::
   "sc_ctrl_inv_rel (Invocations_A.InvokeSchedControlConfigure sc budget period refills badge) sci' =
     (sci' = InvokeSchedControlConfigure sc budget period refills badge)"
 
-(* FIXME RT: preconditions can be reduced, this is what is available at the call site: *)
+lemma decodeSchedContext_Bind_wf:
+  "\<lbrace>valid_cap' (SchedContextCap sc n) and ex_nonz_cap_to' sc
+    and (\<lambda>s. \<forall>cap\<in>set excaps. \<forall>r\<in>zobj_refs' cap. ex_nonz_cap_to' r s)
+    and (\<lambda>s. \<forall>x\<in>set excaps. valid_cap' x s)\<rbrace>
+   decodeSchedContext_Bind sc excaps
+   \<lbrace>valid_sc_inv'\<rbrace>, -"
```

Comment:
Do you mind changing the `sc`'s in these lemmas to something like `scptr`? It's a minor thing, but I find it much easier to follow when it's immediately clear what's a pointer and what's an object.

Reply:
Oh yeah, I normally do do this. I'll change these